File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueCommentDeletedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueCommentDeletedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("commentId")]
    public long CommentId { get; set; }

    [JsonPropertyName("deleterId")]
    public long DeleterId { get; set; }

    [JsonPropertyName("deletedAtUtc")]
    public DateTime DeletedAtUtc { get; set; } = DateTime.UtcNow;
}

File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueCommentDeletedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.IssueComment;

import java.time.Instant;

public record IssueCommentDeletedEvent(
        long projectId,
        long issueId,
        long commentId,
        long deleterId,
        Instant deletedAtUtc
) {
    public static IssueCommentDeletedEvent from(IssueComment comment, long projectId, long deleterId) {
        return new IssueCommentDeletedEvent(
                projectId,
                comment.getIssue().getId(),
                comment.getId(),
                deleterId,
                Instant.now()
        );
    }
}